api/grpc/mpi/v1/command.proto

```diff
@@ -0,0 +1,387 @@
+// Copyright (c) F5, Inc.
+//
+// This source code is licensed under the Apache License, Version 2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+
+// These proto definitions follow https://protobuf.dev/programming-guides/style/
+// and recommendations outlined in https://static.sched.com/hosted_files/kccncna17/ad/2017%20CloudNativeCon%20-%20Mod%20gRPC%20Services.pdf
+syntax = "proto3";
+package mpi.v1;
+
+option go_package = "github.com/nginx/agent/v3/api/grpc/mpi/v1";
+
+import "mpi/v1/common.proto";
+import "mpi/v1/file.proto";
+import "google/protobuf/struct.proto";
+import "google/protobuf/timestamp.proto";
+
+// A service outlining the command and control options for a DataPlane Client
+// All operations are written from a client perspective
+// The RPC calls generally flow Client -> Server, except for Subscribe which contains a bidirectional stream
+// The ManagementPlaneRequest sent in the Subscribe stream triggers one or more client actions.
+// Messages provided by the Management Plane must be a FIFO ordered queue. Messages in the queue must have a monotonically-increasing integer index. 
+// The indexes do not need to be sequential. The index must be a 64-bit signed integer.
+// The index must not reset for the entire lifetime of a unique Agent (i.e. the index does not reset to 0 only because of a temporary disconnection or new session). 
+// Messages must not be removed from the Mangement Plane queue until Ack’d by the Agent. 
+// Messages sent but not yet Ack’d must be kept in an “in-flight” buffer as they may need to be retried.
+service CommandService {
+    // Connects NGINX Agent to the Management Plane agnostic of instance data
+    rpc CreateConnection(CreateConnectionRequest) returns (CreateConnectionResponse) {}
+    // Reports on instances and their configurations
+    rpc UpdateDataPlaneStatus(UpdateDataPlaneStatusRequest) returns (UpdateDataPlaneStatusResponse) {}
+    // Reports on instance health
+    rpc UpdateDataPlaneHealth(UpdateDataPlaneHealthRequest) returns (UpdateDataPlaneHealthResponse) {}
+    // A decoupled communication mechanism between the data plane and management plane.
+    // buf:lint:ignore RPC_RESPONSE_STANDARD_NAME
+    // buf:lint:ignore RPC_REQUEST_STANDARD_NAME
+    rpc Subscribe(stream DataPlaneResponse) returns (stream ManagementPlaneRequest) {}
+}
+
+// The connection request is an intial handshake to establish a connection, sending NGINX Agent instance information
+message CreateConnectionRequest {
+    // Meta-information associated with a message
+    mpi.v1.MessageMeta message_meta = 1;
+    // instance information associated with the NGINX Agent
+    Instance agent = 2;
+}
+
+// This contains a series of NGINX Agent configurations
+message AgentConfig { 
+    // Command server settings
+    Command command = 1;
+    // Metrics server settings
+    Metrics metrics = 2;
```

Comment:
Missing a `File` settings field for configuring the file upload/download endpoint. `self` is a valid value indicating the use of the current gRPC connection, `grpc://` or `https://` endpoints are also valid.